### PR TITLE
Adds a setting to let user assign custom user phrase folder

### PR DESCRIPTION
**Pull request by zonble**
The screenshots are attached.

Screen Shot 2022-05-04 at 12 25 53 AM

Screen Shot 2022-05-04 at 12 25 56 AM

### CODE REVIEW

File: Source/Preferences.swift [hunk 1]
```diff
@@ -401,3 +404,38 @@ class Preferences: NSObject {
...
+
+    @UserDefault(key: kUseCustomUserPhraseLocation, defaultValue: false)
+    @objc static var useCustomUserPhraseLocation: Bool {
+        didSet {
+            if useCustomUserPhraseLocation == false {
+                customUserPhraseLocation = nil
+            } else if customUserPhraseLocation == nil || customUserPhraseLocation == "" {
+                customUserPhraseLocation = UserPhraseLocationHelper.defaultUserPhraseLocation
+            }
+            NotificationCenter.default.post(name: .useCustomUserPhraseLocationDidChange, object: useCustomUserPhraseLocation)
```
Comment (lukhnos) [inline Source/Preferences.swift]: If all the code paths will lead to `.customUserPhraseLocationDidChange` being posted, do we still need `.useCustomUserPhraseLocationDidChange`? Also only the formet gets observed in AppDelegate.

File: Source/PreferencesWindowController.swift [hunk 2]
```diff
@@ -178,6 +185,68 @@ fileprivate let kWindowTitleHeight: CGFloat = 78
...
+    @IBAction func changeUserPhraseEnabledAction(_ sender: Any) {
+        guard let control = sender as? NSControl else {
+            return
+        }
+        let enabled = control.integerValue > 0
+        if enabled {
+            if Preferences.customUserPhraseLocation == nil || Preferences.customUserPhraseLocation == "" {
+                Preferences.customUserPhraseLocation = UserPhraseLocationHelper.defaultUserPhraseLocation
+            } else {
+                Preferences.customUserPhraseLocation = Preferences.customUserPhraseLocation
```
Comment (lukhnos) [inline Source/PreferencesWindowController.swift]: Just to confirm: This is setting the property to its current value. Is this to trigger the posting of  the `.customUserPhraseLocationDidChange` notification?

**Review by lukhnos — APPROVED** [review]
Was just wondering if we can just define one notification, otherwise LGTM!

**Comment by zonble** [comment]
Now I use only one notification to notify app delegate to reload data when changing the location.

**Pull request merged**